multiversx_sdk/abi/serializer.py

```diff
@@ -130,6 +131,9 @@ def _deserialize_counted_variadic_values(self, parts_holder: PartsHolder, value:
             value.items.append(new_item)
             value.length += 1
 
+        if int(length) != value.length:
```

Comment:
Does this work with data such as `... counted variadic, variadic`? A `for` loop as mentioned by Octavian would handle that case.

Reply:
now using a `for` loop